src/transformers/modeling_utils.py

```diff
@@ -2379,8 +2400,10 @@ def save_pretrained(
                     # In the non-tensor case, fall back to the pointer of the object itself
                     ptrs[id(tensor)].append(name)
 
-            # These are all the pointers of shared tensors.
-            shared_ptrs = {ptr: names for ptr, names in ptrs.items() if len(names) > 1}
+            # These are all the pointers of shared tensors, excluding disk offloaded tensors in the meta device
+            shared_ptrs = {
+                ptr: names for ptr, names in ptrs.items() if len(names) > 1 and ptr[0] != torch.device("meta")
+            }
```

Comment:
For `meta` device, we can use `find_tied_parameters` function instead . In from_pretrained, we do 
```python
  if device_map is None and not is_fsdp_enabled() and not is_deepspeed_zero3_enabled():
      ptrs = collections.defaultdict(list)
      for name, tensor in model.state_dict().items():
          id_tensor = id_tensor_storage(tensor)
          ptrs[id_tensor].append(name)

      # These are all the pointers of shared tensors.
      tied_params = [names for _, names in ptrs.items() if len(names) > 1]
  else:
      # id function doesn't work for meta tensor so we need this function
      tied_params = find_tied_parameters(model)
```